polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -1317,6 +1320,33 @@ impl State {
 		self.enable_aggression(network_sender, Resend::No, metrics).await;
 	}
 
+	// When finality is lagging as a last resort nodes start sending the messages they have
+	// multiples times. This means it is safe to accept duplicate messages without punishing the
+	// peer and reduce the reputation and can end up banning the Peer, which in turn will create
+	// more no-shows.
+	fn accept_duplicates_from_validators(
+		blocks_by_number: &BTreeMap<BlockNumber, Vec<Hash>>,
+		topologies: &SessionGridTopologies,
+		aggression_config: &AggressionConfig,
+		entry: &BlockEntry,
+		peer: PeerId,
+	) -> bool {
+		let topology = topologies.get_topology(entry.session);
+		let min_age = blocks_by_number.iter().next().map(|(num, _)| num);
```

Comment:
nit: Don't you want to extract the logic regarding should_trigger_aggression? I see that it's a copy-paste from enable_aggression, and I would feel more at ease if we didn't have to remember to change it in two places in the future.

Reply:
I'll put this as a follow-up item.